lib/pengine/unpack.c

```diff
@@ -1041,11 +1041,15 @@ unpack_node_loop(xmlNode * status, bool fence, pe_working_set_t * data_set)
          state != NULL; state = crm_next_same_xml(state)) {
 
         const char *id = ID(state);
-        const char *uname = NULL;
+        const char *uname = crm_element_value(state, XML_ATTR_UNAME);
         pe_node_t *this_node = NULL;
         bool process = FALSE;
 
-        uname = crm_element_value(state, XML_ATTR_UNAME);
+        if ((id == NULL) || (uname == NULL)) {
+            // Warning already logged in first pass through status section
+            continue;
+        }
+
         this_node = pe_find_node_any(data_set->nodes, id, uname);
 
         if (this_node == NULL) {
```

Comment:
Is this "Node %s is unknown" message redundant (compare to the "Ignoring recorded node status" message in `unpack_status`)? I mention this only because we skip warning on missing id or uname, since those already have warnings in `unpack_status` -- yet we log for unknown node in both places.

Reply:
/me moves ahead to  https://github.com/ClusterLabs/pacemaker/pull/2281/commits/6cba619ecc5d5c3edc26763175f57bff12ab0c95, which modifies the logging so that that both conditions are handled the same.

Never mind :)